### ISSUE TITLE
Failure in configure script

### ISSUE THREAD

**Issue by MrDrMcCoy** [statsite/statsite#257]
# Issue description:
When attempting to compile `statsite`, I receive the following error when running `./configure`:

```
./configure: line 5935: syntax error near unexpected token `CHECK,'
./configure: line 5935: `PKG_CHECK_MODULES(CHECK, check >= 0.9.7, have_check="yes", { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Check not found; cannot run unit tests!" >&5'
```
# Steps to reproduce:

1. Install Ubuntu 14.04 on KVM
2. Install the following packages with apt:
    - autoconf
    - automake
    - build-essential
    - libtool
    - git-core
3. Clone `statsite` repo
4. Run `autogen.sh`
5. Run `configure`

# Workaround:
I was able to get around it by removing the following lines from `configure`, which results in a working makefile and subsequent build:

```
# PKG-Config based checks
PKG_CHECK_MODULES(CHECK, check >= 0.9.7, have_check="yes", { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Check not found; cannot run unit tests!" >&5
$as_echo "$as_me: WARNING: Check not found; cannot run unit tests!" >&2;})
```

**Comment by johnkeates** [comment]
What autotools versions?

**Comment by MrDrMcCoy** [comment]
I have these versions installed:
```
autotools-dev/trusty,now 20130810.1
pkg-config/trusty 0.26-1ubuntu4
```
The environment we have is a bit behind in terms of updates, and uses a private repo mirror.

**Comment by johnkeates** [comment]
It's possible that autotools creates the correct configure script for the build but not for your system due to missing or outdated dependencies. Since it fails at PKG_CHECK it seems to mean to me that the PKG-Config dependencies are too far behind to work with the version of autotools you are using. It might be possible to go to to an even older version of autotools so it expects less from pkg-config. Other than that there isn't much we can do about this.

Could you also verify Check is installed correctly? It should be at least version 0.9. We provide a suitable version in deps, you can manually install that if your repo doesn't have one. It might resolve the problem with PKG Config not finding Check.

**Comment by MrDrMcCoy** [comment]
The available version in our repos is `check/trusty 0.9.10-6ubuntu3`. There is a small but important difference between 'available' and 'installed', and in our case it is not installed. The readme suggests that it is only required for running tests, but further examination of the install document shows that you recommend installing check regardless. 

Closing as user error, sorry to bother you.

**Comment by johnkeates** [comment]
No problem! Also, when building statsite as an installable package (which is possible on Ubuntu, you'll need devscripts and build-essential), the dpkg-buildpackage or debuild commands will want to run the tests for you as well, in which case you'd need check installed too.
